pkg/controller/machineset/controller.go

```diff
@@ -212,11 +214,14 @@ func (r *ReconcileMachineSet) Reconcile(ctx context.Context, request reconcile.R
 			"%s",
 			pausedFalseReason,
 		))
-		machineSet, err := updateMachineSetStatus(r.Client, machineSet, machineSetCopy.Status)
+
+		var err error
+		machineSet, err = updateMachineSetStatus(r.Client, machineSet, machineSetCopy.Status)
```

Comment:
This changes current behaviour, where we now update the `machineSet` passed to `reconcile()` below, rather than losing the update due to the change in scope, which should reduce the likelihood of 409s further on down the line.